cdap-hbase-compat-base/src/main/java/co/cask/cdap/data2/transaction/coprocessor/CConfigurationCache.java

```diff
@@ -84,7 +109,14 @@ public void run() {
             try {
               CConfiguration newCConf = cConfReader.read();
               if (newCConf != null) {
+                Configuration newConf = new Configuration();
```

Comment:
Yes, that is correct. Other option is to check whether there is any change in cConf and then create Configuration object. CConfiguration doesn't override equals method.